x/globalfee/ante/fee.go

```diff
@@ -65,17 +62,38 @@ func (mfd FeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, ne
 		return next(ctx, tx, simulate)
 	}
 
-	// Get required Global Fee and min gas price
-	requiredGlobalFees, err := mfd.getGlobalFee(ctx, feeTx)
+	// Sort fee tx's coins, zero coins in feeCoins are already removed
+	feeCoins := feeTx.GetFee().Sort()
+	gas := feeTx.GetGas()
+	msgs := feeTx.GetMsgs()
+
+	// Get required Global Fee
+	requiredGlobalFees, err := mfd.GetGlobalFee(ctx, feeTx)
 	if err != nil {
 		return ctx, err
 	}
-	requiredFees := GetMinGasPrice(ctx, int64(feeTx.GetGas()))
 
-	// sort fee tx's coins
-	feeCoins := feeTx.GetFee().Sort()
-	gas := feeTx.GetGas()
-	msgs := feeTx.GetMsgs()
+	// Get local minimum-gas-prices
+	localFees := GetMinGasPrice(ctx, int64(feeTx.GetGas()))
+
+	combinedFeeRequirement := CombinedFeeRequirement(requiredGlobalFees, localFees)
+	if len(combinedFeeRequirement) == 0 {
+		return ctx, sdkerrors.Wrapf(sdkerrors.ErrNotFound, "required fees are not setup.")
+	}
+
+	nonZeroCoinFeesReq, zeroCoinFeesDenomReq := splitFees(combinedFeeRequirement)
+
+	// feeCoinsNoZeroDenom is feeCoins after removing the coins whose denom is zero coins' denom in combinedFeeRequirement
```

Comment:
I find this comment confusing and unnecessary -> if you want to include, should read something like:
`contains non-zero denominations from the combinedFeeRequirement`